code/game/objects/effects/mines.dm

```diff
@@ -1,25 +1,117 @@
+
+/obj/item/deployablemine/smartstun
+	name = "deployable smart mine"
+	desc = "An unarmed smart stun mine. It can be planted to arm it."
+	mine_type = /obj/effect/mine/stun/smart
+
+/obj/item/deployablemine/rapid
+	name = "deployable rapid smart mine"
+	desc = "An unarmed smart stun mine designed to be rapidly placeable."
+	mine_type = /obj/effect/mine/stun/smart/adv
+	arming_time = 10
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/deployablemine/heavy
+	name = "deployable sledgehammer smart mine"
+	desc = "An unarmed smart heavy stun mine designed to be hard to disarm."
+	mine_type = /obj/effect/mine/stun/smart/heavy
+	arming_time = 50
+
+/obj/item/deployablemine/shrapnel
+	name = "deployable Sharpnel  mine"
+	desc = "An unarmed sharpnel  mine designed to spray lead everywhere and  be hard to disarm."
+	mine_type = /obj/effect/mine/shrapnel
+	arming_time = 50
+
+/obj/item/deployablemine/explosive
+	mine_type = /obj/effect/mine/explosive
+
+
+/obj/item/deployablemine/gas
+	name = "oxygen gas mine"
+	desc = "An unarmed mine that releases oxygen into the air when triggered. Pretty pointless huh."
+	mine_type = /obj/effect/mine/gas
+
+/obj/item/deployablemine/plasma
+	name = "incendiary mine"
+	desc = "An unarmed mine that releases plasma into the air when triggered, then ignites it."
+	mine_type = /obj/effect/mine/gas/plasma
+
+/obj/item/deployablemine/sleepy
+	name = "knockout mine"
+	desc = "An unarmed mine that releases N2O into the air when triggered. Nighty Night!"
+	mine_type = /obj/effect/mine/gas/n2o
+
+/obj/item/deployablemine/afterattack(atom/plantspot, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(!istype(plantspot,/turf/open)) // you can't plant a mine inside a wall or on a mob
+		return
+
+	if(isspaceturf(plantspot))
+		to_chat(user, "<span class='warning'>you cannot plant a mine in space!</span>")
+		return
+
+	if((istype(plantspot,/turf/open/lava)) || (istype(plantspot,/turf/open/chasm)))
+		to_chat(user, "<span class='warning'>You can't plant the mine here!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You start arming the [src]...</span>")
+	if(do_after(user, arming_time, target = src))
+		new mine_type(plantspot)
+		to_chat(user, "<span class='notice'>You plant and arm the [src].</span>")
+		log_combat(user, src, "planted and armed")
+		qdel(src)
+
 /obj/effect/mine
 	name = "dummy mine"
 	desc = "Better stay away from that thing."
 	density = FALSE
 	anchored = TRUE
 	icon = 'icons/obj/items_and_weapons.dmi'
 	icon_state = "uglymine"
-	/// We manually check to see if we've been triggered in case multiple atoms cross us in the time between the mine being triggered and it actually deleting, to avoid a race condition with multiple detonations
 	var/triggered = FALSE
+	var/smartmine = 0
+	var/disarm_time = 200
+	var/disarm_product = /obj/item/deployablemine // ie what drops when the mine is disarmed
+
+/obj/effect/mine/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/multitool))
```

Comment:
This proc sends a signal at the base. Make it call parent and take the return into account.